Setup rendering of movie screenings for individual movies.

Kookiie88 opened the pull request:
In this update, we've made some big strides in improving both the user experience and the maintainability of our code:

**Divide and Conquer:** 
We've split the original fetch function into two separate functions: fetchMovieScreenings and renderMovieScreenings. This division of labor makes our code easier to digest, test, and keep up. fetchMovieScreenings takes care of getting the movie screenings data from the API, while renderMovieScreenings handles creating and adding list items to the DOM.

**Smooth Sailing with Async:** 
We've transformed the fetch function into an async function, making our promise-based operations a lot more readable and straightforward.

**Date and Time, Upgraded:** 
We're now using the Intl.DateTimeFormat object to format the date and time of the movie screenings. This makes the date and time much more user-friendly.

**From Console to DOM:** 
Instead of just logging the movie screenings data to the console, we're now rendering this data directly to the DOM. For each movie screening, we create a new list item, add the theater and start time to it, and then append this item to the list element in the DOM.

**Screenings, Filtered and Sorted:** 
We've updated the getMovieScreenings function to filter out any past screenings and sort the remaining ones in ascending order by date. This ensures that users only see upcoming screenings that are relevant to them, and that these screenings are presented in a logical order.

**Main Function:** 
A main function was introduced to kick off the script. This function first calls fetchMovieScreenings to get the movie screenings data, and then hands off this data to renderMovieScreenings for display on the webpage. This setup makes it super clear what the script is doing right off the bat, and it's a breeze to modify the script's behavior down the line. Plus, it's a great way to keep things neat and tidy - each function has its own job, so the code is more modular, easier to test, and simpler to maintain.

Kookiie88 commented:
Here is a example of how it looks in the frontend:
![image](https://github.com/Kookiie88/A5_inlamning_3/assets/144999664/bf8371eb-6227-45fe-b2a5-4e4671470459)

AnnaLenaO commented on static/individualScreenings.js:
```diff
@@ -1,10 +1,31 @@
+
+    listItem.innerHTML = `Theater: ${screening.room}<br>Start time: ${time}, ${formattedDate}`;
+    listElement.appendChild(listItem);
   });
```
Is this `});` still supposed to be here after the changes made?

AlexanderK88 replied:
Yes it comes from here
![change](https://github.com/Kookiie88/A5_inlamning_3/assets/144999664/138c387f-6c79-4a63-9cad-c6a89d2b5a60)

AnnaLenaO replied:
Ah, missed that totally! :)

AnnaLenaO commented on static/individualScreenings.js:
```diff
@@ -1,10 +1,31 @@
+    const time = new Intl.DateTimeFormat("default", timeOptions).format(date);
+    const formattedDate = new Intl.DateTimeFormat("default", dateOptions).format(date);
+
+    listItem.innerHTML = `Theater: ${screening.room}<br>Start time: ${time}, ${formattedDate}`;
```
I am not sure but are there alternatives to innerHTML regarding potential saftey issue with innerHTML?

AlexanderK88 replied:
Ahh good catch i will change it to use textContent of innerHTML instead to avoid potential XSS attacks.

AnnaLenaO reviewed (commented):
As you mention in the description there are informative names for the functions and one gets a clear picture of what the code achives. Very nice!

Kookiie88 commented:
I changed so the code is not using innerHTML anymore but instead textContent and i also added classes to both "p" elements so i can style them later.

![image](https://github.com/Kookiie88/A5_inlamning_3/assets/144999664/41669794-7591-4188-a130-2654fe0f583d)

MatsJoh50 reviewed (approved):
I cant find anything to comment about, it looks like the description in both browser and in code.

great job.
